test/smoke/tests.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package smoke
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+
+	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/jetstack/cert-manager/test/unit/gen"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cmpapi "github.com/cert-manager/policy-approver/pkg/apis/policy/v1alpha1"
+)
+
+var _ = Describe("Smoke", func() {
+	It("should create a CertificateRequestPolicy, RBAC bind to all users, deny and approver a request", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		cl, err := client.New(cnf.RestConfig, client.Options{
+			Scheme: cmpapi.GlobalScheme,
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating test Namespace")
+		namespace := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "smoke-test-policy-",
+			},
+		}
+		Expect(cl.Create(ctx, &namespace)).NotTo(HaveOccurred())
+
+		By("Creating test SelfSigned Issuer")
+		issuer := cmapi.Issuer{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "smoke-test-policy-",
+				Namespace:    namespace.Name,
+			},
+			Spec: cmapi.IssuerSpec{IssuerConfig: cmapi.IssuerConfig{SelfSigned: &cmapi.SelfSignedIssuer{}}},
+		}
+		Expect(cl.Create(ctx, &issuer)).NotTo(HaveOccurred())
```

Comment:
So because we don't test that a cert gets issued we therefore don't need to wait for the Issuer to become Ready 👍🏼